src/app/Library/CrudPanel/Traits/FakeFields.php

```diff
@@ -12,26 +13,50 @@ trait FakeFields
      * plus the '_token' and 'redirect_after_save' variables.
      *
      * @param  array  $requestInput  The request input.
-     * @param  string  $form  The CRUD form. Can be 'create' or 'update' . Default is 'create'.
-     * @param  int|bool  $id  The CRUD entry id in the case of the 'update' form.
+     * @param  array  $fields
      *
      * @see \Illuminate\Http\Request::all() For an example on how to get the request input.
      *
      * @return array The updated request input.
      */
-    public function compactFakeFields($requestInput)
+    public function compactFakeFields($requestInput, $model = false, $fields = [])
     {
         // get the right fields according to the form type (create/update)
-        $fields = $this->fields();
+        if (empty($fields)) {
+            $fields = $this->fields();
+        }
+
+        $model = $model ? (is_string($model) ? app($model) : $model) : $this->model;
 
         $compactedFakeFields = [];
+
         foreach ($fields as $field) {
             // compact fake fields
             // cast the field name to array first, to account for array field names
             // in fields that send multiple inputs and want them all saved to the database
             foreach ((array) $field['name'] as $fieldName) {
-                if (isset($field['fake']) && $field['fake'] == true && array_key_exists($fieldName, $requestInput)) {
-                    $fakeFieldKey = isset($field['store_in']) ? $field['store_in'] : 'extras';
+                $fakeFieldKey = isset($field['store_in']) ? $field['store_in'] : 'extras';
+                $isFakeField = $field['fake'] ?? false;
+
+                // field is represented by the subfields
+                if (isset($field['subfields']) && $field['model'] === get_class($model)) {
+                    foreach ($field['subfields'] as $subfield) {
+                        $subfieldName = Str::afterLast($subfield['name'], '.');
+                        $isSubfieldFake = $subfield['fake'] ?? false;
+                        $subFakeFieldKey = isset($subfield['store_in']) ? $subfield['store_in'] : 'extras';
+
+                        if (array_key_exists($subfieldName, $requestInput) && $isSubfieldFake) {
+                            $this->addCompactedField($requestInput, $subfieldName, $subFakeFieldKey);
+
+                            if (! in_array($subFakeFieldKey, $compactedFakeFields)) {
+                                $compactedFakeFields[] = $subFakeFieldKey;
+                            }
+                        }
+                    }
+                    continue;
+                }
```

Comment:
This is probably not the most ideal way to do this - a recursive function would probably clean this up. But for now, it'll do, since IT WORKS A OK!